shared/actions/fs/index.js

```diff
@@ -575,21 +575,25 @@ const editSuccess = (res, action, state: TypedState) => {
 const editFailed = (res, {payload: {editID}}) => Saga.put(FsGen.createEditFailed({editID}))
 
 function* openPathItem(action: FsGen.OpenPathItemPayload): Saga.SagaGenerator<any, any> {
-  const {path, routePath} = action.payload
+  const {path, routePath, openDirectly} = action.payload
```

Comment:
I think instead of mixing `openDirectly` into the existing action, it might be better to have a new action, e.g. `openPathItemDirectly`, for two reasons:

1) It'd help keep action handlers simple, and we wouldn't need the if-conditions below;

2) This depends on `state.fs.pathItems`. When the action is dispatched from within the Files tab, it's guaranteed we've already loaded the needed data. But if we are going from the widget, we might not have them yet. So we need a way to load those first (probably just with a stat call). It'd become convoluted if we try to add it here.

WDYT?

Reply:
1. SGTM, I'll make the change.
2. Right now you're correct, but this won't be true going forward, since we'll be making the necessary calls as part of retrieving history.

Reply:
Cool; thanks!

> Right now you're correct, but this won't be true going forward, since we'll be making the necessary calls as part of retrieving history.

Hmm when we retrieve the history, don't we still need to check `pathItems` to know if a thing is `folder`? I guess I don't know the details enough that I might be imagining something up.

Reply:
@songgao I was envisioning that the saga that retrieves history ensures that the relevant path items are updated.